android/app/src/main/java/app/organicmaps/util/PowerManagerCompat.java

```diff
@@ -0,0 +1,109 @@
+package app.organicmaps.util;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Build;
+import android.os.PowerManager;
+import android.provider.Settings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+public class PowerManagerCompat
+{
+  private final String TAG = PowerManagerCompat.class.getSimpleName();
+  private final Map<String, Integer> POWER_SAVE_MODE_VALUES = new HashMap<String, Integer>()
+  {
+    {
+      put("HUAWEI", 4);
+      put("XIAOMI", 1);
+    }
+  };
+  private final ArrayList<String> POWER_SAVE_MODE_SETTINGS_NAMES = new ArrayList<String>()
+  {
+    {
+      add("SmartModeStatus"); // huawei setting name
+      add("POWER_SAVE_MODE_OPEN"); // xiaomi setting name
+    }
+  };
+  private final ArrayList<String> POWER_SAVE_MODE_CHANGE_ACTIONS = new ArrayList<String>()
+  {
+    {
+      add("huawei.intent.action.POWER_MODE_CHANGED_ACTION");
+      add("miui.intent.action.POWER_SAVE_MODE_CHANGED");
+    }
+  };
+
+  public void monitorPowerSaveModeChanged(Context context, PowerSaveModeChangeListener listener)
+  {
+    if (POWER_SAVE_MODE_VALUES.containsKey(Build.MANUFACTURER.toUpperCase(Locale.getDefault())))
+    {
+      context.registerReceiver(new BroadcastReceiver()
```

Comment:
yes for now its of no use. should i remove the implementation of the same from PowerManagerCompat class? All the broadcast receivers and everything except the method for checking the status?

Reply:
Thanks for the feedback. This issue is probably because broadcast reciever unregister is called when no broadcast receiver was registered. I guess i placed unregister call in wrong method of mwmActivty.

I will check it.

Reply:
The issue was with unregister method actually i was not unregistering the broadcast reciever properly. Now i have created a variable and saved the instance into it.